diff --git a/src/PMFRG_xyz.jl b/src/PMFRG_xyz.jl
index 8cf9984..4778ae2 100644
--- a/src/PMFRG_xyz.jl
+++ b/src/PMFRG_xyz.jl
@@ -10,6 +10,7 @@ using SpinFRGLattices,
 using SpinFRGLattices.StaticArrays
 using Unroll
 using MuladdMacro
+using LIKWID
 using FastBroadcast
 
 setZero!(a::AbstractArray{T,N}) where {T,N} = fill!(a, zero(T))
@@ -1206,7 +1207,7 @@ function getXBubble!(
                     iuh_end = min(iuh_start + iuh_blocksize - 1, iuh_max)
                     block_length = iuh_end - iuh_start + 1
 
-                    addY!(
+                    @marker "addY!" addY!(
                         Buffs.X_sum_addY,
                         Gamma,
                         Workspace.Par.System,
@@ -1222,7 +1223,7 @@ function getXBubble!(
 
                     ### If no u--t symmetry, then add all the bubbles
                     ### If use u--t symmetry, then only add for nu smaller then nt (all other obtained by symmetry)
-                    addX!(
+                    @marker "addX!" addX!(
                         Buffs.X_sum_addX,
                         Gamma,
                         Workspace.Par.System,
@@ -1428,17 +1429,19 @@ end
 
 using JLD2
 function getDeriv!(Deriv, State, setup, Lam; saveArgs = true)
+    Marker.startregion("getDeriv!")
 
     (; X, Par) = setup # use pre-allocated X and XTilde to reduce garbage collector time
     Workspace = OneLoopWorkspace(State, Deriv, X, Par)
 
     getDFint!(Workspace, Lam)
     get_Self_Energy!(Workspace, Lam)
-    getXBubble!(Workspace, Lam)
+    @marker "getXBubble!" getXBubble!(Workspace, Lam)
     symmetrizeBubble!(Workspace.X, Par)
     addToVertexFromBubble!(Workspace.Deriv.Gamma, Workspace.X)
     symmetrizeVertex!(Workspace.Deriv.Gamma, Par)
 
+    Marker.stopregion("getDeriv!")
     return
 end
 
@@ -1551,7 +1554,7 @@ function testPMFRG!(State, setup, Deriv!::Function; loadArgs = false)
 end
 
 SolveFRG(Par, isotropy; kwargs...) =
-    launchPMFRG!(InitializeState(Par, isotropy), AllocateSetup(Par), getDeriv!; kwargs...)
+    @marker "launchPMFRG!" launchPMFRG!(InitializeState(Par, isotropy), AllocateSetup(Par), getDeriv!; kwargs...)
 TestFRG(Par, isotropy; kwargs...) =
     testPMFRG!(InitializeState(Par, isotropy), AllocateSetup(Par), getDeriv!; kwargs...)
 
