diff --git a/src/PMFRG_xyz.jl b/src/PMFRG_xyz.jl
index 2ea20d4..30cd935 100644
--- a/src/PMFRG_xyz.jl
+++ b/src/PMFRG_xyz.jl
@@ -7,6 +7,7 @@ module PMFRG_xyz

 using RecursiveArrayTools
 using SpinFRGLattices, OrdinaryDiffEq, DiffEqCallbacks, RecursiveArrayTools, StructArrays
+using Recorder
 using SpinFRGLattices.StaticArrays
 using Unroll

@@ -990,7 +991,7 @@ function getDeriv!(Deriv, State, setup, Lam; saveArgs = true)

     getDFint!(Workspace, Lam)
     get_Self_Energy!(Workspace, Lam)
-    getXBubble!(Workspace, Lam)
+    @record 1:5:50 getXBubble!(Workspace, Lam)
     symmetrizeBubble!(Workspace.X, Par)
     addToVertexFromBubble!(Workspace.Deriv.Gamma, Workspace.X)
     symmetrizeVertex!(Workspace.Deriv.Gamma, Par)
