diff --git a/src/PMFRG_xyz.jl b/src/PMFRG_xyz.jl
--- a/src/PMFRG_xyz.jl
+++ b/src/PMFRG_xyz.jl
@@ -7,6 +7,7 @@ module PMFRG_xyz

 using SpinFRGLattices, OrdinaryDiffEqLowOrderRK, DiffEqCallbacks, StructArrays
 using Unroll
+using Recorder
 using MuladdMacro
 using FastBroadcast

@@ -1036,7 +1037,7 @@ function getDeriv!(Deriv, State, setup, Lam; saveArgs = true)

     getDFint!(Workspace, Lam)
     get_Self_Energy!(Workspace, Lam)
-    getXBubble!(Workspace, Lam)
+    @record 1:5:50 getXBubble!(Workspace, Lam)
     symmetrizeBubble!(Workspace.X, Par)
     addToVertexFromBubble!(Workspace.Deriv.Gamma, Workspace.X)
     symmetrizeVertex!(Workspace.Deriv.Gamma, Par)
